exo/inference/inference_engine.py

```diff
@@ -29,6 +29,11 @@ async def infer_prompt(self, request_id: str, shard: Shard, prompt: str, inferen
     output_data = await self.infer_tensor(request_id, shard, tokens, inference_state)
     return output_data 
 
+inference_engine_classes = {
```

Comment:
this is tricky and seems likely to break at some point. Ideally, every inference engine would have a decorator, so that at startup, we can register all of them somewhere, and then use that variable/registry. This is gonna break when a new inference engine is added and someone forgets to update this.

As a shorter term compromise though, I'd at least reference the class itself and grab `__name__`, so that if someone refactors a class name, this doesn't break?

Reply:
The reason this is hairy is because on platforms that don't support mlx the import breaks

Reply:
So like, this is a hack, but it's a hack that needs more extensive work to work around, for the same reason get_inference_engine does

Reply:
what I'm imagining is a decorator such as

```
def inference_engine(name):
    def decorator(cls):
        inference_engine_registry[name] = cls
        return cls
    return decorator
```

(`inference_engine_registry` would live in inference/inference_engine.py with the abstract parent class)

Then, every engine would have something like this:

```
@inference_engine("mlx")
class MLXDynamicShardInferenceEngine(InferenceEngine):
    # ...
```

When someone is building a new inference engine, they are likely to look at the structure of others, so it'll be more obvious that it needs the decorator.

Reply:
I think that's a good design, but it doesn't seem to me like it'd solve the fundamental problem, which is that you can't import the class definition at all if your platform doesn't support the underlying framework